tpu-client-next/src/connection_workers_scheduler.rs

```diff
@@ -0,0 +1,121 @@
+//! This module defines [`ConnectionWorkersScheduler`] which creates and
+//! orchestrates `ConnectionWorker` instances.
+use {
+    super::{leader_updater::LeaderUpdater, SendTransactionStatsPerAddr},
+    crate::{
+        connection_worker::ConnectionWorker,
+        quic_networking::{
+            create_client_config, create_client_endpoint, QuicClientCertificate, QuicError,
+        },
+        transaction_batch::TransactionBatch,
+        workers_cache::{WorkerInfo, WorkersCache, WorkersCacheError},
+    },
+    log::*,
+    quinn::Endpoint,
+    solana_sdk::signature::Keypair,
+    std::{net::SocketAddr, sync::Arc},
+    thiserror::Error,
+    tokio::sync::mpsc,
+};
+
+/// Size of the channel to transmit transaction batches to the target workers.
+const WORKER_CHANNEL_SIZE: usize = 2;
```

Comment:
Is the assumption that we expect the tx generator/scheduler to be the bottleneck? Or do we ever expect to block on submitting tx batches to the workers?

Reply:
The bottleneck was always sending over network part because server limits number of streams/receive window size to quite limiting values. As far as I remember generator can easily generate 350k tps but the sender part will be limited by ~4k tps in the best scenario (with one connection, one stake, no fanout).

Reply:
Followup, but this (and possibly others) needs to be arguments to some call. The
client shouldn't make assumptions about at what rate txs are going to be produced.

Reply:
I will introduce a configuration structure with this and some other parameters to be specified